app/lib/katello/event_daemon/runner.rb

```diff
@@ -7,6 +7,7 @@ class Runner
 
       class << self
         def initialize
```

Comment:
I'm reading through old PRs trying to see how intentional this was. I found this: https://github.com/Katello/katello/pull/8403

In it:

"""
The flow is like this:

    Worker A starts serving a request and checks to see if the Daemon is already started, if so, it does nothing and continues startup
    If the daemon is not running, a lock is placed on an arbitrary file, and daemon startup begins, writing its pidfile
    If Worker B serves a request at the same time and also finds the daemon not started it waits for the file lock to be released (it's imperceptible since daemon start is fast)
    Worker B sees that the daemon is already started by Worker A, and continues as usual

"""

It seems the lock being placed when the daemon is not running was indeed on purpose.

Reply:
But `touch(lock_file)` doesn't lock anything, it only ensures the file is present. The actual locking happens in `start`:
https://github.com/Katello/katello/blob/4950081967a99de4b68825cbe86ea8845334b155/app/lib/katello/event_daemon/runner.rb#L54

So I'd ultimately think that `initialize` is either not needed *or* buggy (I tend to lean towards not needed) ;-)